### ISSUE TITLE
pull_session with url paramters isn't working

### ISSUE THREAD

**Issue by ericman93** [bokeh/bokeh#7785]
I'm trying to call `pull_session` from my flask app.
bokeh version is `0.12.15`

```
session = pull_session(url=f'http://localhost:5006/my_app?plot=scatter')
```

The bokeh server is processing the request normaly, but i'm having this error
```
OSError: Cannot pull session document because we failed to connect to the server (to start the server, try the 'bokeh serve' command)
```
Without the url parameters everything is working great

_Bokeh server_
```
from tornado.ioloop import IOLoop
from bokeh.server.server import Server
from bokeh.layouts import widgetbox, row
from bokeh.models.widgets import Slider
from bokeh.models import ColumnDataSource
from bokeh.plotting import figure
import numpy as np


def sliders_modify_doc(curdoc):
    # Set up data
    N = 200
    x = np.linspace(0, 4 * np.pi, N)
    y = np.sin(x)
    source = ColumnDataSource(data=dict(x=x, y=y))

    # Set up widgets
    offset = Slider(title="offset", value=0.0, start=-5.0, end=5.0, step=0.1)
    amplitude = Slider(title="amplitude", value=1.0, start=-5.0, end=5.0, step=0.1)
    phase = Slider(title="phase", value=0.0, start=0.0, end=2 * np.pi)
    freq = Slider(title="frequency", value=1.0, start=0.1, end=5.1, step=0.1)

    # Set up plot
    plot = figure(plot_height=400, plot_width=400, title="my sine wave",
                  tools="crosshair,pan,reset,save,wheel_zoom",
                  x_range=[0, 4 * np.pi], y_range=[-2.5, 2.5])

    plot.line('x', 'y', source=source, line_width=3, line_alpha=0.6)

    def update_data(attrname, old, new):
        # Get the current slider values
        a = amplitude.value
        b = offset.value
        w = phase.value
        k = freq.value

        # Generate the new curve
        x = np.linspace(0, 4 * np.pi, N)
        y = a * np.sin(k * x + w) + b

        source.data = dict(x=x, y=y)

    for w in [offset, amplitude, phase, freq]:
        w.on_change('value', update_data)

    inputs = widgetbox(offset, amplitude, phase, freq)
    curdoc.add_root(row(inputs, plot, width=800))
    
def start_bokeh_worker():
    server = Server({'/my_app': sliders_modify_doc},
                    io_loop=IOLoop(),
                    allow_websocket_origin=["localhost:5000"])

    server.start()
    server.io_loop.start()


if __name__ == '__main__':
    start_bokeh_worker()
```

_Flask server_
```
from flask import Flask, render_template
from bokeh.client import pull_session
from bokeh.embed import server_session

app = Flask(__name__, template_folder='templates')

@app.route('/bokeh')
def get():
    session = pull_session(url=f'http://localhost:5006/my_app?plot=scatter')
    script = server_session(None, session.id, url='http://localhost:5006/my_app')

    return render_template("bokeh.html", script=script, template="Flask")

if __name__ == '__main__':
    app.run()
```
_bokeh.html_
```
<html>
    <body>
      {{ script|safe }}
    </body>
</html>
```

Also, the last parameter has a suffix (when no parameters are given `request.arguments` is empty
![image](https://user-images.githubusercontent.com/5768348/38633130-64aff7f0-3dc7-11e8-8ce9-a672dd8203e8.png)

**Comment by bryevdv** [comment]
@ericman93 the issue template asked for quite a bit more information, including versions, and a minimal complete reproducer. Can you provide this information? If not, we will have to close this as `norepro`

**Comment by ericman93** [comment]
Just updated to issue :+1:

**Comment by bryevdv** [comment]
Thanks, still needing bokeh version information

**Comment by ericman93** [comment]
bokeh version is `0.12.15` and the code is fixed :+1:

**Comment by ericman93** [comment]
I was able to fix it by calling `server_document` instead of `pull_session` and then `server_session`

**Comment by bryevdv** [comment]
@ericman93 thanks this was reproducible, which means is should be possible to do something about it. Guessing it was just overlooking in the other PR.

**Comment by DunkeeFunk** [comment]
Im struggling with similar issues. I hope they fix this one soon too

**Comment by bryevdv** [comment]
This is resolved in #8242, but arguments must be provided as a dict, as elsewhere:
```
session = pull_session(url='http://localhost:5006/foo', arguments=dict(plot='scattaer'))
```
It also cannot be stressed enough, arguments can *only have effect when pulling **new** sessions* If a session already exists, then the code that might do something with the arguments has already run and gone.
